ergotree-interpreter/src/sigma_protocol/verifier.rs

```diff
@@ -392,5 +402,48 @@ mod tests {
                 prop_assert_eq!(ver_res.unwrap().result, true, "verify failed on secret: {:?}", &secret);
             }
         }
+
+        #[test]
+        fn test_prover_verifier_atleast(secret1 in any::<DlogProverInput>(),
```

Comment:
@greenhat Does this test look OK? If the implementation was sound then it should pass?

Still having troubles figuring out why verification fails

Reply:
The test looks fine. Yes, it should.
There is a test with test vectors for `AtLeast` in Scala - https://github.com/ScorexFoundation/sigmastate-interpreter/blob/795c176debfe105646a7deef3f01bd537b79c81d/sigmastate/src/test/scala/sigmastate/crypto/SigningSpecification.scala#L119-L135
Could you implement it and then we'll see if our verifier is correct and the problem is in the prover.
We have corresponding tests for similar test vectors already in https://github.com/ergoplatform/sigma-rust/blob/9b5ca5a9e4ccbc076991c9db349645c4db848cae/ergotree-interpreter/tests/signing_spec_tests.rs#L325-L393

Reply:
I implemented this in a previous PR: https://github.com/ergoplatform/sigma-rust/blob/c308476ad074147596b03fb9f3dcb04018823a72/ergotree-interpreter/tests/signing_spec_tests.rs#L400-L466

I totally blanked on this, so that narrows it down to the prover

Reply:
Sorry, I missed it. 
I feel you. Hunting for the error in the prover sucks. I appreciate you're taking care of this!